mono/metadata/assembly.c

```diff
@@ -558,6 +558,23 @@ mono_assembly_getrootdir (void)
 }
 
 /**
+ * mono_native_getrootdir:
+ * 
+ * Obtains the root directory used for looking up native libs (.so, .dylib).
+ *
+ * Returns: a string with the directory, this string should not be freed.
+ */
+G_CONST_RETURN gchar *
+mono_native_getrootdir (void)
+{
+	gchar* fullpath = g_malloc(strlen(mono_assembly_getrootdir ()) + strlen(mono_config_get_reloc_lib_dir()) + 1);
+	strcpy( fullpath, mono_assembly_getrootdir () );
+	strcat( fullpath, "/" ) ;
+	strcat( fullpath, mono_config_get_reloc_lib_dir() ) ;
+	return fullpath;
```

Comment:
I think this can be replaced with

```c
gchar* fullpath = g_build_path (G_DIR_SEPARATOR_S, mono_assembly_getrootdir (), mono_config_get_reloc_lib_dir(), NULL);
return fullpath;
```

fullpath needs to be `g_free()`'ed at the call site (also fix the doc comment to mention this).